src/modules/jcms_admin/jcms_admin.module

```diff
@@ -193,3 +250,55 @@ function _jcms_admin_node_title_unique($title, $node, $content_type = NULL) {
 
   return TRUE;
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function jcms_admin_entity_subqueue_presave(EntityInterface $entity) {
+  if ($entity->bundle() != 'covers_preview') {
+    return NULL;
+  }
+  if (_jcms_admin_covers_active('publish')) {
+    $items = ($entity->get('items')) ? $entity->get('items')->getValue() : [];
+    $items = array_splice($items, 0, $entity->get('field_covers_active_items')->getString());
+    foreach ($items as $item) {
+      $node = Node::load($item['target_id']);
+      $moderation_info = \Drupal::service('content_moderation.moderation_information');
+      if (!$moderation_info->isLatestRevision($node)) {
+        $latest = $moderation_info->getLatestRevision($node->getEntityTypeId(), $node->id());
+        $latest->moderation_state->target_id = 'published';
+        $latest->save();
+      }
+      else {
+        $node->moderation_state->target_id = 'published';
+        $node->save();
+      }
+    }
+    $subqueue = EntitySubqueue::load('covers');
+    $subqueue->set('items', $items);
+    drupal_set_message(\Drupal::translation()->formatPlural(count($items),
+      '%number active item has been published to the current covers list from preview list.',
+      '%number active item(s) have been published to the current covers list from preview list.',
+      [
+        '%number' => count($items),
+      ]));
+    $subqueue->save();
+  }
+  if (_jcms_admin_covers_active('reset')) {
+    $subqueue = EntitySubqueue::load('covers');
+    $items = ($subqueue->get('items')) ? $subqueue->get('items')->getValue() : [];
+    $entity->set('field_covers_active_items', 3);
+    foreach ($items as $item) {
+      $node = Node::load($item['target_id']);
+      $node->moderation_state->target_id = 'published';
```

Comment:
I looked in the `content_moderation` module for a constant for this value and couldn't find one. I think it's a good idea but it really should belong in that module.